source/simulator/checkpoint_restart.cc

```diff
@@ -469,31 +416,34 @@ namespace aspect
 #endif
 
     // Then start with the actual deserialization.
+
+    const std::string path = parameters.output_directory + "restart/" + Utilities::int_to_string(last_checkpoint_id, 2) + "/";
```

Comment:
```suggestion
    const std::string checkpoint_path = parameters.output_directory + "restart/" + Utilities::int_to_string(last_checkpoint_id, 2) + "/";
```

or `restart_path`, but something that indicates what type of path this is.